flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -224,7 +225,11 @@ private static HiveConf createHiveConf(
         } else {
             hadoopConf = getHadoopConfiguration(hadoopConfDir);
         }
-        if (hadoopConf == null) {
+        if (hadoopConf == null && !isNullOrWhitespaceOnly(hadoopConfDir)) {
+            throw new CatalogException(
+                    "Failed to load the hadoop conf from specified path:" + hadoopConfDir,
+                    new FileNotFoundException("Path " + hadoopConfDir + " does not exist."));
```

Comment:
Another situation is `isNullOrWhitespaceOnly` exists but none of the conf files exist in that folder. We should throw exception for that too.

Reply:
@lirui-apache , ok. 
Additional,  try to  find the paths with configuration files as far as possible from possiable hadoop conf path.